src/server/common/dbconn.cpp

```diff
@@ -153,25 +153,25 @@ std::unique_ptr<simple_wml::document> dbconn::get_game_history(int player_id, in
 "  else 'trunk' "
 "  end as VERSION "
 "from "+db_game_player_info_table_+" player , "+db_game_content_info_table_+" scenario , "+db_game_content_info_table_+" era , "+db_game_info_table_+" game ";
-	// using a left join when searching by modification won't exclude anything
-	game_history_query += search_content_type == 2 && !search_content.empty() ? "inner join " : "left join ";
-	game_history_query += db_game_content_info_table_+" mods "
-"   on mods.TYPE = 'modification' "
-"  and mods.INSTANCE_UUID = game.INSTANCE_UUID "
-"  and mods.GAME_ID = game.GAME_ID ";
 	// modification id optional parameter
-	if(search_content_type == 2)
+	if(search_content_type == 2 && !search_content.empty())
 	{
-		if(!search_content.empty())
-		{
-			game_history_query += "and mods.ID like ? ";
+		game_history_query += ", "+db_game_content_info_table_+" mods "
+"  where mods.TYPE = 'modification' "
+"  and mods.INSTANCE_UUID = game.INSTANCE_UUID "
+"  and mods.GAME_ID = game.GAME_ID "
+"  and mods.ID like ? ";
 
-			utils::to_sql_wildcards(search_content, false);
-			params.emplace_back(search_content);
-		}
+		utils::to_sql_wildcards(search_content, false);
+		params.emplace_back(search_content);
+	}
+	else
+	{
+		// put the where clause with an always true condition here so I don't need to check again a couple lines down whether it needs "where" vs "and"
+		game_history_query += "where 1=1 ";
```

Comment:
I don't know for sure, but from what I understand, SQL databases effectively compile every statement into a "query plan" before running them. This process can potentially rewrite the entire form of the query, so I would suspect that it's likely also able to recognize an always true condition and eliminate it.

This is mostly speculation, however.

Also, even if I'm right, that may not imply there's no performance impact – maybe there could still be an impact on the "building a query plan" stage.

Reply:
I've used it with other databases for the same reason as here and haven't seen any impact there either. It would be incredibly bizarre for a query optimizer to not be able to optimize away something like `1=1` though, and even if it didn't I can't imagine how actually executing that comparison in the query plan would take anywhere close to a meaningful amount of time.

Reply:
Seems silly to do this just to save a trivial check in C++ though.

Reply:
It's a trivial check, but I'd still prefer to only need to do it on one place rather than two.

Reply:
Well, can't say I understand but I guess I don't mind. I would rather use `1` or `true` then though.

Reply:
What do you mean by "1 or true"? It's already comparing 1=1.

Reply:
What for? Just `1` or `true` does the same thing just shorter and IMO more readable?

Reply:
Oh, I didn't know that `where true` is valid SQL.